ckan/logic/action/update.py

```diff
@@ -317,10 +318,11 @@ def package_update(
         model.Session.rollback()
         raise ValidationError(errors)
 
-    #avoid revisioning by updating directly
-    model.Session.query(model.Package).filter_by(id=pkg.id).update(
-        {"metadata_modified": datetime.datetime.utcnow()})
-    model.Session.refresh(pkg)
+    if not resources_only:
+        #avoid revisioning by updating directly
+        model.Session.query(model.Package).filter_by(id=pkg.id).update(
```

Comment:
I think we should add the  .with_for_update so it locks the record for the update.
like
https://github.com/ckan/ckan/blob/81b56c55e5e3651d7fcf9642cd5a489a9b62212c/ckan/model/package.py#L145

i.e.
 model.Session.query(model.Package).with_for_update().filter_by(id=pkg.id).update(